lib/location/method_gnss.c

```diff
@@ -412,16 +422,9 @@ bool method_gnss_agps_required(struct nrf_modem_gnss_agps_data_frame *request)
 #endif
 
 #if defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_NRF_CLOUD_PGPS)
-static void method_gnss_request_assistance(void)
+static void method_gnss_assistance_request(void)
```

Comment:
This may not be the correct place but I think we need to have a pseudo-algorithm on how A-GPS (and maybe P-GPS) data retrieval works with all throttling, LTE connection status, RRC connection release, PSM etc. Now it's documented but scattered around this file so it's hard to follow. This could be documented somewhere in this file (a single place whether it's perfect or not), library documentation (maybe too detailed stuff in there), or somewhere else internally. We can discuss about this.

Reply:
Yes, that's probably needed. I'm currently running tests with A-GPS + P-GPS and I'm already thinking about improving the logic with P-GPS. So, it's probably going to get even more complicated. Not sure if we have other potential places than here in the C file or in the library documentation.

Reply:
Yes, I agree. pseudoalg would be appreciated. Also, there are a lot of compile time ifdefs and was thinking if all configuration options are documented? I.e what could be compiled on at the same time? AGPS/PGPS_EXTERNAL cannot be configured on at the same time as CONFIG_NRF_CLOUD_PGPS/CONFIG_NRF_CLOUD_AGPS. SO, wondering should we have also build assert for those?

Reply:
Actually, `CONFIG_NRF_CLOUD_AGPS` and `CONFIG_NRF_CLOUD_PGPS` are always needed for A-GPS and P-GPS support. The same nRF Cloud library implementation is used also with `CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL` and `CONFIG_LOCATION_METHOD_GNSS_PGPS_EXTERNAL`.

I'll come up with some kind of pseudo code/algorithm for the logic, because it's not that obvious how it all works.

Reply:
OK, thanks